app/server/models/token.py

```diff
@@ -43,5 +63,27 @@ def token_amount_to_system(self, token_amount):
         return int(token_amount) * 100 / 10**self.decimals
 
     def system_amount_to_token(self, system_amount):
-        return int(float(system_amount) / 100 * 10**self.decimals)
+        return int(float(system_amount) * 10**self.decimals / 100)
+
+    def add_exchange_contract(self, exchange_contract: ExchangeContract):
+
+        exchange_transfer_account = (server.models.transfer_account.TransferAccount.query
+                                     .filter_by(token=self)
+                                     .filter_by(exchange_contract=exchange_contract)
+                                     .first())
+
+        if not exchange_transfer_account:
+            exchange_transfer_account = server.models.transfer_account.TransferAccount(
```

Comment:
huh? why would this exist if the previous one didn't?

Reply:
I had a case when I was dev-ing this, though tbh it's probably not something that would occur in general without something really weird happening. Should I just remove?